### DISCUSSION TITLE
Restrict editing of IP addresses to specific prefix

### DISCUSSION THREAD

**Discussion by clerie** [netbox-community/netbox#13118] [Help Wanted!]
I want to allow specific users editing IP address objects within some prefixes. So naturally I would add a permission with something like the following constraint:

```json
{ "prefix__tenant__slug": "mycustomer" }
```

But there is no relationship between IP address objects and prefixes. How could I achieve such a behaviour instead?

**Comment by CTV-2023** [comment]
With specific tags on these IP maybe ?
Specific permission applied to the user/a group granting "change" on "ipam | IP address" with constraint
{"tags__name": "mycustomer"}

**Reply by clerie** [reply]
I could also just do:

```
{ "tenant__slug": " mycustomer" }
```

But that would still allow creating IP addresses within any prefix as long as it has the specified tenant assigned.

It is important for me to restrict the prefix for which IP addresses can be created so they don't end up in our backbone configs by accident.

**Reply by CTV-2023** [reply]
Did you try with a read only custom field applied on the prefix ? I don't know if you can use a constraint on them

**Reply by clerie** [reply]
How would that read only custom field help restricting the creation of IP address objects within a specific prefix only?

**Reply by CTV-2023** [reply]
Forget customfields, settings permissions for them seems to be complicated.
I made another test with a tenant called "mycustomer". Assign the IP address to it
For my user profile, I chose a readonly permission for everything
And a custom permission applied on "ipam | IP address" allowing to "change" and with constraint {"tenant__slug": "mycustomer"}	
Of course if you need your customer to be able to create IP addresses this is more complex

**Reply by clerie** [reply]
The mentioned constraint is fine for viewing IP address objects, but for write permission this would allow creating any IP address as long as it is referencing the tenant. This is bad because they could create IP addresses in our backbone range which is bad.

**Reply by CTV-2023** [reply]
Yes, that's why I didn't allow "create", only "change" existing IP addresses. I don't think your use case is currently possible in Netbox (pure RBAC)

**Reply by clerie** [reply]
As far as I know, permissions are per object. But you can change the IP address of the IP address object afterwards and it keeps the object id. That means you can't allow editing for IP addresses at all.

Do you have any idea how to implement this instead of adding an additional foreign key?
